server/webapp/WEB-INF/rails/webpack/models/new_pipeline_configs/pipeline_config.ts

```diff
@@ -54,9 +54,9 @@ export class PipelineConfig extends ValidatableMixin {
     this.validatePresenceOf("group");
     this.validateIdFormat("group");
 
-    this.materials = Stream(new NameableSet(materials));
-    this.validateNonEmptyCollection("materials", {message: `A pipeline must have at least one material`});
-    this.validateAssociated("materials");
+    this.materials = Stream(new Materials(...materials));
+    // this.validateNonEmptyCollection("materials", {message: `A pipeline must have at least one material`});
+    // this.validateAssociated("materials");
```

Comment:
So for `validateAssociated` to work, the associated object needs to extend `ValidatableMixin`.

In this case `Materials` class already extends `Array` class. To extend `ValidatableMixin` as well, we need to use `applyMixin`. This will copy properties from `ValidatableMixin` to `Materials` class.

But somehow this was not working. I was repeatedly was getting error 'this.__errors is not a function'

My guess is, it was because of `Object.setPrototypeOf(this, Object.create(Materials.prototype));` this line in Materials constructor.

Reply:
I extended from Array class because we followed that convention in other models as well. We can update `Materials` class to not extend and array.